src/vs/editor/contrib/colorPicker/browser/colorHoverParticipant.ts

```diff
@@ -225,28 +224,35 @@ function renderHoverParts(participant: ColorHoverParticipant | StandaloneColorPi
 }
 
 function _updateEditorModel(editor: ICodeEditor, range: Range, model: ColorPickerModel, context?: IEditorHoverRenderContext) {
+
+	const edit = model.presentation.textEdit;
+	const textEdits = [edit ?? { range, text: model.presentation.label, forceMoveMarkers: false }];
+	if (model.presentation.additionalTextEdits) {
+		textEdits.push(...model.presentation.additionalTextEdits);
+	}
+	editor.executeEdits('colorpicker', textEdits);
+	editor.pushUndoStop();
+
 	let newRange: Range;
-	const textEdits: ISingleEditOperation[] = [];
-	if (model.presentation.textEdit) {
-		textEdits.push(model.presentation.textEdit);
+	if (edit) {
+		const separatedText = edit.text.split('\n');
+		const startLineNumber = edit.range.startLineNumber;
+		const startColumnNumber = edit.range.startColumn;
+		const endLineNumber = edit.range.endLineNumber + separatedText.length - 1;
+		const endColumnNumber = endLineNumber === startLineNumber ?
+			startColumnNumber + separatedText[separatedText.length - 1].length :
+			separatedText[separatedText.length - 1].length;
 		newRange = new Range(
-			model.presentation.textEdit.range.startLineNumber,
-			model.presentation.textEdit.range.startColumn,
-			model.presentation.textEdit.range.endLineNumber,
-			model.presentation.textEdit.range.endColumn
+			startLineNumber,
+			startColumnNumber,
+			endLineNumber,
+			endColumnNumber
 		);
 		const trackedRange = editor.getModel()!._setTrackedRange(null, newRange, TrackedRangeStickiness.GrowsOnlyWhenTypingAfter);
```

Comment:
I don't know what _setTrackedRange does, but it could be an issue that the edits are now applied before this call and not afterwards.